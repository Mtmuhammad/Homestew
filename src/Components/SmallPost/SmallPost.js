import React from "react";
import "./SmallPost.scss";
import { Link } from "react-router-dom";

const SmallPost = ({ id, title, img }) => {
  return (
    <div className="col-6 col-lg-3 mb-5">
      <Link to={`/recipes/${id}`}>
        <article className="text-center small-post">
          <div className="small-post-img">
            <img
              src={img || "./No-image.svg"}
              alt={title}
              className="img-fluid"
            />
          </div>
          <div className="small-post-content mt-4">
            <h4 className="quaternary">{title}</h4>
          </div>
        </article>
      </Link>
    </div>
  );
};

export default SmallPost;
