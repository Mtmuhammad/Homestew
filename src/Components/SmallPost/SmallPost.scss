@use "../../Styles/abstracts/mixins" as *;
@use "../../Styles/abstracts/variables" as *;

.small-post {
  &-img img {
    height: 250px;
    border-radius: 5px;
    transition: all 0.2s;

    @include respond(sport) {
      height: 200px;
    }
    @include respond(sphone) {
      height: 180px;
    }
    @include respond(tiny) {
      height: 160px;
    }

    &:hover {
      box-shadow: 0px 0px 13px 7px rgba(151, 151, 151, 0.552);
      transform: scale(1.01);
    }
  }

  &-content {
    font-size: 2.2rem;
  }

  & h4 {
    transition: all 0.3s;
  }
}
a {
  text-decoration: none;

  &:hover h4 {
    color: $primary-green;
  }
}
