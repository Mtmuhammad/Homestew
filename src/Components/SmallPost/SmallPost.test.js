import React from "react";
import { render, screen } from "@testing-library/react";
import SmallPost from "./SmallPost";
import { BrowserRouter } from "react-router-dom";

describe("smoke and snapshot tests", () => {
  test("should render without crashing", () => {
    render(
      <BrowserRouter>
        <SmallPost />
      </BrowserRouter>
    );
  });

  test("should match snapshot", () => {
    const { asFragment } = render(
      <BrowserRouter>
        <SmallPost />
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});

describe("should render with props", () => {
  test("should render content", () => {
    render(
      <BrowserRouter>
        <SmallPost
          id={123}
          title={"Chicken Burgers"}
          image={"chickenBurgers.jpg"}
        />
      </BrowserRouter>
    );
    const article = screen.getByRole("article");
    expect(article).toBeInTheDocument();
    expect(article).toHaveClass("small-post");

    const img = screen.getByAltText(/chicken burgers/i);
    expect(img).toBeInTheDocument();

    const heading = screen.getByRole("heading");
    expect(heading).toBeInTheDocument();
    expect(heading).toHaveClass("quaternary");
  });
});
