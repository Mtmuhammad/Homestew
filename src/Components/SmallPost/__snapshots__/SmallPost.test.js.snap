// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`smoke and snapshot tests should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="col-6 col-lg-3 mb-5"
  >
    <a
      href="/recipes/undefined"
    >
      <article
        class="text-center small-post"
      >
        <div
          class="small-post-img"
        >
          <img
            class="img-fluid"
            src="./No-image.svg"
          />
        </div>
        <div
          class="small-post-content mt-4"
        >
          <h4
            class="quaternary"
          />
        </div>
      </article>
    </a>
  </div>
</DocumentFragment>
`;
